src/components/ContextMenu/ContextMenu.jsx

```diff
@@ -0,0 +1,228 @@
+		 */
+		className: string,
+		/**
+		 * Passed through to the root element.
+		 */
+		style: object,
+		/**
+		 * Indicates whether the FlyOut will render or not. Defaults to `true`.
+		 */
+		isExpanded: bool,
+		/**
+		 * alignment of the FlyOut relative to Target. Defaults to `'down'`.
+		 */
+		direction: oneOf(['down', 'up']),
+		/**
+		 * The `id` of the FlyOut portal element that is appended to `document.body`. Defaults to a generated `id`.
+		 */
+		portalId: string,
+		/**
+		 * Called once when one of following happens:
+		 *
+		 * - Called with `BELOW_FOLD` if direction is `down` and FlyOut clips screen bottom.
+		 * - Called with `ABOVE_FOLD` if direction is `up` and it has enough space to render down without clipping.
+		 */
+		onChangeBounds: func,
+		/**
+		 * Called when a click event happenens outside of the ContextMenu.
+		 */
+		onClickOut: func
+	},
+
+	childProps: {
+		Target: {},
+		FlyOut: {
+			style: object
+		}
+	},
+
+	getDefaultProps() {
+		return {
+			isExpanded: true,
+			direction: 'down',
+			onChangeBounds: null,
+			onClickOut: null
+		};
+	},
+
+	getInitialState() {
+		const {
+			portalId,
+			direction
+		} = this.props;
+		return {
+			portalId: portalId || 'ContextMenu-Portal-' + Math.random().toString(16).substr(2),
+			targetRect: {
+				bottom: 0,
+				top: 0,
+				left: 0,
+				right: 0,
+				height: 0,
+				width: 0
+			},
+			flyOutHeight: 0,
+			isFlyoutBelowFold: direction !== ContextMenu.DOWN
+		};
+	},
+
+	componentDidMount() {
+		let targetDOMNode = this.refs.target;
+
+		this.updateTargetRectangleIntervalId = setInterval(() => {
+			this.alignFlyOut(targetDOMNode);
+		}, 10);
+
+		this.onClickBodyEventListener = window.addEventListener('click', (event) => {
+			if (this.props.onClickOut && this.refs.flyOutPortal) {
+				let flyOutEl = this.refs.flyOutPortal.portalElement.firstChild;
+				if (!(flyOutEl.contains(event.target) || targetDOMNode.contains(event.target))) {
+					this.props.onClickOut(event);
+				}
+			}
+		});
+	},
+
+	componentWillUnmount() {
+		clearInterval(this.updateTargetRectangleIntervalId);
+		window.document.body.removeEventListener('click', this.onClickBodyEventListener);
+	},
+
+	componentWillReceiveProps() {
+		let targetDOMNode = this.refs.target;
+
+		if (targetDOMNode) {
+			this.alignFlyOut(targetDOMNode);
+		}
+	},
+
+	statics: {
+		DOWN: 'down',
+		UP: 'up',
+		BELOW_FOLD: 'BELOW_FOLD',
+		ABOVE_FOLD: 'ABOVE_FOLD',
+		isElementBelowFold(targetElement, yPosition) {
+			if (typeof window !== 'undefined') {
+				let bodyRect = window.document.body.getBoundingClientRect();
+				let targetElementRect = targetElement.getBoundingClientRect();
+				yPosition = (yPosition === undefined ? targetElementRect.top : yPosition);
+				if (yPosition + targetElementRect.height - bodyRect.top > window.innerHeight - bodyRect.top) {
+					return true;
+				} else {
+					return false;
+				}
+			} else {
+				throw new Error('Browser only! Cannot access window object.');
+			}
```

Comment:
Since the `if` clause always returns can we eliminate the use of an `else` block?